src/stream/mod.rs

```diff
@@ -1051,6 +1053,19 @@ pub trait Stream {
     {
         split::split(self)
     }
+
+    /// Converts a Stream<Item=T,Error=E> into a `Stream<Item=Result<T,E>>`.
+    ///
+    /// The returned `Stream` will never return an `Err` on poll.
+    /// Instead, the item type is a result, matching the result
+    /// of polling the original stream.
+    ///
+    /// This is useful if the stream is attached to a sink,
+    /// and the sink should handle any errors rather than the
+    /// stream itself.
+    fn results<E>(self) -> Results<Self, E> where Self: Sized {
```

Comment:
Could this PR hold off on adding this for now? Perhaps this could remain private for just here?